snapcraft/internal/parts.py

```diff
@@ -117,6 +124,25 @@ def get_part(self, part_name, full=False):
                 remote_part.pop(key)
         return remote_part
 
+    def matches_for(self, part_match, max_len=0):
+        sm = difflib.SequenceMatcher(isjunk=None, autojunk=False)
+        sm.set_seq2(part_match)
```

Comment:
using sm as a variable is a gopher thing. Something like "matcher" would be more pythonic.